core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -76,6 +76,8 @@ class BlockManagerMasterEndpoint(
 
   // Mapping from block id to the set of block managers that have the block.
   private val blockLocations = new JHashMap[BlockId, mutable.HashSet[BlockManagerId]]
+  private val visibleRDDBlocks = new mutable.HashSet[RDDBlockId]
+  private val tidToRddBlockIds = new mutable.HashMap[Long, mutable.HashSet[RDDBlockId]]
```

Comment:
When do we clean up this if a task fails?

Reply:
Thanks, fixed. When task completes, will clean up the records.